diff -ru llvm-5.0.1.src/include/llvm/IR/Instructions.h llvm-5.0.1.src.patched/include/llvm/IR/Instructions.h
--- llvm-5.0.1.src/include/llvm/IR/Instructions.h	2017-07-11 16:23:00.000000000 -0600
+++ llvm-5.0.1.src.patched/include/llvm/IR/Instructions.h	2018-04-01 14:26:55.197380688 -0600
@@ -4195,11 +4195,11 @@
   }
 
 public:
-  using DerefFnTy = std::pointer_to_unary_function<Value *, BasicBlock *>;
+  using DerefFnTy = BasicBlock *(*)(Value *);
   using handler_iterator = mapped_iterator<op_iterator, DerefFnTy>;
   using handler_range = iterator_range<handler_iterator>;
   using ConstDerefFnTy =
-      std::pointer_to_unary_function<const Value *, const BasicBlock *>;
+      const BasicBlock *(*)(const Value *);
   using const_handler_iterator =
       mapped_iterator<const_op_iterator, ConstDerefFnTy>;
   using const_handler_range = iterator_range<const_handler_iterator>;
diff -ru llvm-5.0.1.src/utils/TableGen/CodeGenDAGPatterns.cpp llvm-5.0.1.src.patched/utils/TableGen/CodeGenDAGPatterns.cpp
--- llvm-5.0.1.src/utils/TableGen/CodeGenDAGPatterns.cpp	2017-06-27 01:10:20.000000000 -0600
+++ llvm-5.0.1.src.patched/utils/TableGen/CodeGenDAGPatterns.cpp	2018-04-01 12:34:17.860889841 -0600
@@ -239,7 +239,7 @@
   TypeSet InputSet(*this);
 
   // Filter out all the fp types.
-  TypeVec.erase(remove_if(TypeVec, std::not1(std::ptr_fun(isInteger))),
+  TypeVec.erase(remove_if(TypeVec, std::not_fn(isInteger)),
                 TypeVec.end());
 
   if (TypeVec.empty()) {
@@ -264,7 +264,7 @@
   TypeSet InputSet(*this);
 
   // Filter out all the integer types.
-  TypeVec.erase(remove_if(TypeVec, std::not1(std::ptr_fun(isFloatingPoint))),
+  TypeVec.erase(remove_if(TypeVec, std::not_fn(isFloatingPoint)),
                 TypeVec.end());
 
   if (TypeVec.empty()) {
@@ -290,7 +290,7 @@
   TypeSet InputSet(*this);
 
   // Filter out all the vector types.
-  TypeVec.erase(remove_if(TypeVec, std::not1(std::ptr_fun(isScalar))),
+  TypeVec.erase(remove_if(TypeVec, std::not_fn(isScalar)),
                 TypeVec.end());
 
   if (TypeVec.empty()) {
@@ -314,7 +314,7 @@
   bool MadeChange = false;
 
   // Filter out all the scalar types.
-  TypeVec.erase(remove_if(TypeVec, std::not1(std::ptr_fun(isVector))),
+  TypeVec.erase(remove_if(TypeVec, std::not_fn(isVector)),
                 TypeVec.end());
 
   if (TypeVec.empty()) {
